plotly_resampler/aggregation/aggregation_interface.py

```diff
@@ -1,172 +1,334 @@
 """AbstractSeriesAggregator interface-class, subclassed by concrete aggregators."""
 
+from __future__ import annotations
+
 __author__ = "Jonas Van Der Donckt"
 
 import re
 from abc import ABC, abstractmethod
-from typing import List, Tuple
+from typing import List, Optional, Tuple
 
 import numpy as np
-import pandas as pd
-
 
-class AbstractSeriesAggregator(ABC):
-    """"""
 
+class AbstractAggregator(ABC):
     def __init__(
         self,
         interleave_gaps: bool = True,
-        nan_position: str = "end",
-        dtype_regex_list: List[str] = None,
+        x_dtype_regex_list: Optional[List[str]] = None,
+        y_dtype_regex_list: Optional[List[str]] = None,
```

Comment:
will look into this!

Reply:
I woudl leave this as is for now; because:
1) tsdownsample it's interface has a hard lock-in on returning the selected indices; which is only true for the `DataPointSelector` classes.
2) And since, we also support data aggregation e.g. mean / ...; serving a dtype regex to that function might be beneifical.

Main thing still left to do:
- [ ] look if there is any unnecessary overhead w.r.t. the dtype regex (e.g. we define them for both the `plotly-resampler` and `tsdownsample` interface.) -> maybe we can just reuse `tsdownsample` its `interface for the `datapoint-selector` class?